pkg/analysis/passes/screenshots/screenshots.go

```diff
@@ -39,24 +44,59 @@ func checkScreenshotsExist(pass *analysis.Pass) (interface{}, error) {
 		explanation := "Screenshots are displayed in the Plugin catalog. Please add at least one screenshot to your plugin.json."
 		pass.ReportResult(pass.AnalyzerName, screenshots, "plugin.json: should include screenshots for the Plugin catalog", explanation)
 		return data.Info.Screenshots, nil
-	} else {
-		reportCount := 0
-		for _, screenshot := range data.Info.Screenshots {
-			if strings.TrimSpace(screenshot.Path) == "" {
-				reportCount++
-				pass.ReportResult(pass.AnalyzerName, screenshots, fmt.Sprintf("plugin.json: invalid empty screenshot path: %q", screenshot.Name), "The screenshot path must not be empty.")
-			}
-		}
+	}
 
-		if reportCount > 0 {
-			return nil, nil
+	reportCount := 0
+	for _, screenshot := range data.Info.Screenshots {
+		if strings.TrimSpace(screenshot.Path) == "" {
+			reportCount++
+			pass.ReportResult(pass.AnalyzerName, screenshots, fmt.Sprintf("plugin.json: invalid empty screenshot path: %q", screenshot.Name), "The screenshot path must not be empty.")
+		} else if !validImageType(screenshot.Path) {
```

Comment:
Is it better to do this after `invalid empty screenshot` in a separate loop? Added here so we can use the same loop.